spec/java/util/LinkedList.automata.lsl

```diff
@@ -0,0 +1,1236 @@
+    }
+
+
+    fun *.pop (@target self: LinkedList): Object
+    {
+        result = _unlinkFirst();
+    }
+
+
+    fun *.push (@target self: LinkedList, e: Object): void
+    {
+        _linkAny(0, e);
+    }
+
+
+    fun *.remove (@target self: LinkedList): Object
+    {
+        result = _unlinkFirst();
+    }
+
+
+    fun *.remove (@target self: LinkedList, o: Object): boolean
+    {
+        result = _unlinkByFirstEqualsObject(o);
+    }
+
+
+    fun *.remove (@target self: LinkedList, index: int): Object
+    {
+        _checkElementIndex(index);
+        result = _unlinkAny(index);
+    }
+
+
+    // within java.util.AbstractCollection
+    fun *.removeAll (@target self: LinkedList, c: Collection): boolean
+    {
+        action TODO();
+    }
+
+
+    fun *.removeFirst (@target self: LinkedList): Object
+    {
+        result = _unlinkFirst();
+    }
+
+
+    fun *.removeFirstOccurrence (@target self: LinkedList, o: Object): boolean
+    {
+        result = _unlinkByFirstEqualsObject(o);
+    }
+
+
+    // within java.util.Collection
+    fun *.removeIf (@target self: LinkedList, filter: Predicate): boolean
+    {
+        action TODO();
+    }
+
+
+    fun *.removeLast (@target self: LinkedList): Object
+    {
+        if (this.size == 0)
+            action THROW_NEW("java.util.NoSuchElementException", []);
+
+        result = _unlinkAny(this.size - 1);
+    }
+
+
+    fun *.removeLastOccurrence (@target self: LinkedList, o: Object): boolean
+    {
+        val index: int = action LIST_FIND(this.storage, o, 0, this.size);
+        if (index == -1)
+        {
+            result = false;
+        }
+        else
+        {
+            result = true;
+
+            // there should be no elements to the right of the previously found position
+            val nextIndex: int = index + 1;
+            if (nextIndex < this.size)
+            {
+                val rightIndex: int = action LIST_FIND(this.storage, o, nextIndex, this.size);
+                action ASSUME(rightIndex == -1);
+            }
+
+            // actual removal and associated modifications
+            action LIST_REMOVE(this.storage, index);
+            this.size -= 1;
+            this.modCount += 1;
+        }
+    }
+
+
+    // within java.util.List
+    fun *.replaceAll (@target self: LinkedList, operator: UnaryOperator): void
+    {
+        action TODO();
+    }
+
+
+    // within java.util.AbstractCollection
+    fun *.retainAll (@target self: LinkedList, c: Collection): boolean
+    {
+        action TODO();
+    }
+
+
+    fun *.set (@target self: LinkedList, index: int, element: Object): Object
+    {
+        _checkElementIndex(index);
+        action LIST_SET(this.storage, index, element);
+        result = action LIST_GET(this.storage, index);
+    }
+
+
+    fun *.size (@target self: LinkedList): int
+    {
+        result = this.size;
+    }
+
+
+    // within java.util.List
+    fun *.sort (@target self: LinkedList, c: Comparator): void
```

Comment:
I have same question about sort like in ArrayList...

Reply:
Fixed.